src/Sylius/Bundle/ResourceBundle/Form/EventSubscriber/AddCodeFormSubscriber.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ResourceBundle\Form\EventSubscriber;
+
+use Sylius\Component\Resource\Model\CodeAwareInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Intl\Exception\UnexpectedTypeException;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class AddCodeFormSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @return array
+     */
+    public static function getSubscribedEvents()
+    {
+        return array(
+            FormEvents::PRE_SET_DATA => 'preSetData',
+        );
+    }
+
+    /**
+     * @param FormEvent $event
+     */
+    public function preSetData(FormEvent $event)
+    {
+        $resource = $event->getData();
+
+        if (!$resource instanceof CodeAwareInterface) {
```

Comment:
@michalmarcinkowski Good point, I think it will be ok in current and new version, but we will see. Fortunately, if that's the case, then it is a change in a single place. :)